packages/flutter/lib/src/widgets/platform_view.dart

```diff
@@ -581,6 +581,129 @@ class _UiKitPlatformView extends LeafRenderObjectWidget {
   }
 }
 
+/// The parameters used to create a [PlatformViewController].
+///
+/// See also [CreatePlatformView] which uses this object to create a [PlatformViewController].
+class PlatformViewCreationParams {
+
+  const PlatformViewCreationParams._({
+    @required this.id,
+    @required this.onPlatformViewCreated}):assert(id != null),
+                          assert(onPlatformViewCreated != null);
+
+  /// The auto generated id for the newly created platform view.
+  ///
+  /// [PlatformViewController.viewId] should match this id.
+  final int id;
+
+  /// Notifies when the PlatformView is ready.
+  ///
+  /// Must be invoked as soon as the embedded platform view is ready.
+  /// For example, on iOS, this has to be called when we know the `UIView` is initialized.
+  final PlatformViewCreatedCallback onPlatformViewCreated;
+}
+
+/// The factory to construct a [PlatformViewSurface].
+///
+/// It is required when constructing a [PlatformViewLink].
+typedef PlatformViewSurfaceFactory = PlatformViewSurface Function(BuildContext context, PlatformViewController controller);
+
+/// Constructs a [PlatformViewController].
+///
+/// The implementer of a new platform view is responsible to implement this method when constructing a [PlatformViewLink].
+typedef CreatePlatformView = PlatformViewController Function(PlatformViewCreationParams params);
+
+/// The widget responsible for handling the life cycle and focus for a platform view.
```

Comment:
Links a platform view with the Flutter framework.

Provides common functionality for embedding a platform view (e.g an `android.view.View` on Android) with the Flutter framework.

{@macro flutter.widgets.platformViews.lifetime}